apis/namespaced/apigee/v1beta1/zz_targetserver_types.go

```diff
@@ -0,0 +1,316 @@
+	Port *float64 `json:"port,omitempty" tf:"port,omitempty"`
+
+	// Immutable. The protocol used by this TargetServer.
+	// Possible values are: HTTP, HTTP2, GRPC_TARGET, GRPC, EXTERNAL_CALLOUT.
+	Protocol *string `json:"protocol,omitempty" tf:"protocol,omitempty"`
+
+	// Specifies TLS configuration info for this TargetServer. The JSON name is sSLInfo for legacy/backwards compatibility reasons -- Edge originally supported SSL, and the name is still used for TLS configuration.
+	// Structure is documented below.
+	SSlInfo *SSlInfoInitParameters `json:"sSlInfo,omitempty" tf:"s_sl_info,omitempty"`
+}
+
+type TargetServerObservation struct {
+
+	// A human-readable description of this TargetServer.
+	Description *string `json:"description,omitempty" tf:"description,omitempty"`
+
+	// The Apigee environment group associated with the Apigee environment,
+	// in the format organizations/{{org_name}}/environments/{{env_name}}.
+	EnvID *string `json:"envId,omitempty" tf:"env_id,omitempty"`
+
+	// The host name this target connects to. Value must be a valid hostname as described by RFC-1123.
+	Host *string `json:"host,omitempty" tf:"host,omitempty"`
+
+	// an identifier for the resource with format {{env_id}}/targetservers/{{name}}
+	ID *string `json:"id,omitempty" tf:"id,omitempty"`
+
+	// Enabling/disabling a TargetServer is useful when TargetServers are used in load balancing configurations, and one or more TargetServers need to taken out of rotation periodically. Defaults to true.
+	IsEnabled *bool `json:"isEnabled,omitempty" tf:"is_enabled,omitempty"`
+
+	// The port number this target connects to on the given host. Value must be between 1 and 65535, inclusive.
+	Port *float64 `json:"port,omitempty" tf:"port,omitempty"`
+
+	// Immutable. The protocol used by this TargetServer.
+	// Possible values are: HTTP, HTTP2, GRPC_TARGET, GRPC, EXTERNAL_CALLOUT.
+	Protocol *string `json:"protocol,omitempty" tf:"protocol,omitempty"`
+
+	// Specifies TLS configuration info for this TargetServer. The JSON name is sSLInfo for legacy/backwards compatibility reasons -- Edge originally supported SSL, and the name is still used for TLS configuration.
+	// Structure is documented below.
+	SSlInfo *SSlInfoObservation `json:"sSlInfo,omitempty" tf:"s_sl_info,omitempty"`
+}
+
+type TargetServerParameters struct {
+
+	// A human-readable description of this TargetServer.
+	// +kubebuilder:validation:Optional
+	Description *string `json:"description,omitempty" tf:"description,omitempty"`
+
+	// The Apigee environment group associated with the Apigee environment,
+	// in the format organizations/{{org_name}}/environments/{{env_name}}.
+	// +crossplane:generate:reference:type=github.com/upbound/provider-gcp/apis/namespaced/apigee/v1beta1.Environment
+	// +crossplane:generate:reference:extractor=github.com/crossplane/upjet/v2/pkg/resource.ExtractResourceID()
+	// +kubebuilder:validation:Optional
```

Comment:
This was the subject of the above comments about XRM compliancy. Looks like the reference injector has done its job because there exists this example manifest:
```
resource "google_apigee_target_server" "apigee_target_server" {
  name        = "my-target-server"
  description = "Apigee Target Server"
  protocol    = "HTTP"
  host        = "abc.foo.com"
  port        = 8080
  env_id      = google_apigee_environment.apigee_environment.id
}
```

For consistency, we may consider adding the references for the other resources too. 

But we should also check if it's really possible to reference an env group instead of an env. If so, I think it's better to remove this cross-resource reference here in the target server resource.